tests/draft4/refRemote.json

```diff
@@ -70,5 +70,35 @@
                 "valid": false
             }
         ]
+    },
+    {
+        "description": "Change of scope during ref resolution",
+        "schema": {
+            "definitions": {
+                "schema1": {
+                    "id": "http://localhost:1234/subSchemas.json",
+                    "a": {"$ref": "#/integer"},
+                    "integer": {
+                        "type": "integer"
+                    }
+                }
+            },
+            "integer": {
+                "type": "string"
+            },
+            "$ref": "#/definitions/schema1/a"
```

Comment:
You cannot have a `$ref` together with other members like `definitions` and `integer`. See [the JSON Reference RFC](http://tools.ietf.org/html/draft-pbryan-zyp-json-ref-03#section-3)
```
Any members other than "$ref" in a JSON Reference object SHALL be ignored.
```